gemini/sample-apps/photo-discovery/README.md

```diff
@@ -0,0 +1,81 @@
+To learn more about the subject of the photo, tap “tell me more” to chat with Khanh, an AI agent build built with Vertex AI Agent Builder, and ask follow-up questions about the history of subject in the photo based on information Wikipedia.
+
+The agent can also identify merchandise from the [Google Merchandise Store](https://googlemerchandisestore.com/) and provide product name, description, pricing, and purchase link by referencing a Google Merchandise store dataset.
+
+> [!NOTE]
+Check out the Google I/O 2024 talk for a full walkthrough: [Build generative AI agents with Vertex AI Agent Builder and Flutter](https://youtu.be/V8P_S9OLI_I?si=N2QMBs7HNZL6mKU0).
+
+## Demo
+![Chat UI - Lake Wakatipu](https://github.com/khanhnwin/flutter-vertex-agent/assets/12416400/29a23152-d29b-4083-8589-fe4751331381)
+
+## Getting Started
+
+### Building & deploying the agent with Vertex AI Agent Builder
+
+1. Create a search data store on Vertex AI Search: See [Create a search data store](https://cloud.google.com/generative-ai-app-builder/docs/create-data-store-es) to create your own search data store. The "Data store ID" will be your search engine ID.
+
+1. Deploy a Cloud Run app: Edit /ag-web/app/app.py and /ag-web/app/deploy.sh and set the project ID, GCS bucket name and the search engine ID. Run deploy.sh to deploy the Run app. Open "ag-web" app on the Cloud Run console, and find the hostname (eg "ag-web-xxxxxx.a.run.app").
+
+1. Deploy a Reasoning Engine agent: Open [Cloud Workbench](https://cloud.google.com/vertex-ai/docs/workbench/instances/create-console-quickstart) and upload /ag-web/ag_setup_re.ipynb. Open the Notebook and edit the GOOGLE_SHOP_VERTEXAI_SEARCH_URL with the Run hostname. Run the Notebook from the start to deploy the agent to the Reasoning Engine runtime. From the output on the deployment, find the reasoning engine ID. The output format is "projects/PROJECT_ID/locations/LOCATION/reasoningEngines/REASONING_ENGINE_ID".
+
+1. Redeploy the Cloud Run app: Edit /ag-web/app/app.py and set the REASONING_ENGINE_ID. Run /ag-web/app/deploy.sh to redeploy it.
+
+### Running the Flutter App
+
+1. Ensure that you have [Flutter set up](https://flutter.dev/get-started) on your machine.
+   
+1. Flutter enables building multiplaform apps, so this app has been built to be run on iOS, Android, web, and desktop. Make sure to install any preferred target platform (aka where you want to run the app) requirements such as the iOS Simulator, Android Emulator, an Android phone, Chrome browser, etc.)
+
+1. Change directory into the Flutter project using `cd app`
+
+1. Set up a Firebase project and connect it to this Flutter app by following [step 1 in these instructions.](https://firebase.google.com/docs/vertex-ai/get-started?platform=flutter) Only complete Step 1! Don't add the Vertex AI for Firebase Dart SDK because this source code already adds it as a dependency. 
+
+1. Update the variable CloudRunHost in `app/lib/config.dart` with your Cloud Run endpoint host.
```

Comment:
I don't see this file within this PR, is it generated by one of these steps? Or missing from the PR?

Reply:
Resolved by force pushing `lib` directory overriding root .gitignore.